src/Interpreters/executeQuery.cpp

```diff
@@ -346,9 +350,38 @@ QueryLogElement logQueryStart(
 
         if (elem.type >= settings.log_queries_min_type && !settings.log_queries_min_query_duration_ms.totalMilliseconds())
         {
-            if (auto query_log = context->getQueryLog())
-                query_log->add(elem);
+            if (!settings.log_query_settings && settings.log_query_settings.changed)
+                LOG_TRACE(
+                    getLogger("executeQuery"),
+                    "Query settings will not be added to query_log since setting `log_query_settings` has been set to false."
+                    " The setting has been changed for the query");
+
+            query_log->add(elem);
+        }
+        else if (elem.type < settings.log_queries_min_type)
+        {
+            if (settings.log_queries_min_type.changed)
+                LOG_TRACE(
+                    getLogger("executeQuery"),
+                    "Query start record will not be added to query_log due to query type is bigger than `log_queries_min_type` setting."
```

Comment:
```suggestion
                    "Query start record will not be added to query_log because the query type is smaller than `log_queries_min_type` setting"
```